### ISSUE TITLE
support spreadding an array after and before explicit function parameters

### ISSUE THREAD

**Issue by heyimtcn** [vlang/v#22394]
### Describe the feature

allow this:
```v
const foo = [3,4,5]

fn bar(baz ...int) {
	dump(baz)
}

fn main() {
	bar(1,2,...foo)
	bar(...foo,6,7)
}
```

### Use Case

needed by eptx on discord (https://discord.com/channels/592103645835821068/592294828432424960/1290834184905953403)

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### Version used

V 0.4.8 2b546c4

### Environment details (OS name and version, etc.)

V full version: V 0.4.8 18eee34.2b546c4
OS: windows, Microsoft Windows 11 Pro v22631 64-bit
Processor: 16 cpus, 64bit, little endian, 

vexe: C:\Users\nicof\Desktop\v\v.exe
vexe mtime: 2024-10-02 19:33:15

vroot: OK, value: C:\Users\nicof\Desktop\v
VMODULES: OK, value: C:\Users\nicof\.vmodules
VTMP: OK, value: C:\Users\nicof\AppData\Local\Temp\v_0

Git version: git version 2.45.2.windows.1
Git vroot status: weekly.2024.34-260-g2b546c4f
.git/config present: true

CC version: Error: 'cc' is not recognized as an internal or external command,
operable program or batch file.

thirdparty/tcc: N/A

**Comment by islonely** [comment]
I don't think this should be allowed. Just stumbling across this in some code 9 times out of 10 you would probable assume the `1, 2,` is separate function parameters from `...foo`. It would be difficult to read and understand what's happening without knowing in advanced that the underlying function uses variable number of arguments. But also, what situation can you not already do this:
```v
mut foo := [1, 2]
foo << [3, 4, 5]
bar(...foo)
```

**Comment by medvednikov** [comment]
Yeah let's keep it readable and simple.
